tests/virt_autotest/switch_to_ssh_and_install_hypervisor.pm

```diff
@@ -49,7 +49,7 @@ sub run {
     die 'Need one of both to be true: is_kvm_host || is_xen_host' unless is_kvm_host || is_xen_host;
     my $hypervisor = is_kvm_host ? 'kvm' : 'xen';
     zypper_call('--gpg-auto-import-keys ref');
-    zypper_call("in -t pattern ${hypervisor}_server ${hypervisor}_tools", 1800);
+    script_retry("zypper -n in -t pattern ${hypervisor}_server ${hypervisor}_tools", timeout => 1800, retry => 5, delay => 10);
```

Comment:
This may increase the possibility to pass this step of pkg installation, but the job time will be much longer, which is non-negligible sacrifice.

@okurz Hi Oliver, we are not very familiar with O3 infra setup. Can you give some help here? Is it known unstable/badly network performed when installing packages from tumbleweed repo on O3? Or are we actually not using the best way to do such operations? How do you recommend?

Reply:
Oliver is aware that download.opensuse.org is unstable and showed me two PRs via slack. https://suse.slack.com/archives/C02CANHLANP/p1674025447904969

From our side, what we can do is to make test more resilient by retry or other methods. 

Retry is helpful in some cases , for example we can retry zypper. But I fail to find a good way figure out the download failure in 'virt-install' command, fg. https://openqa.opensuse.org/tests/3096451#step/unified_guest_installation/296

Reply:
Let me provide the references here directly. For example there is https://github.com/openSUSE/zypper/issues/399 which describes not the same symptoms but maybe is linked to the same root cause mentioning https://progress.opensuse.org/issues/123172. There is also https://github.com/openSUSE/zypper/issues/420.

In the meantime https://progress.opensuse.org/issues/123172 was resolved (on 2023-01-26) so I suggest if you still see problems then make sure there actually is a corresponding error report for the corresponding compoment, e.g. in https://progress.opensuse.org/projects/opensuse-admin/issues

Reply:
@Julie-CAO Based on oliver's above suggestion, do you think we should open a ticket for it in https://progress.opensuse.org/projects/opensuse-admin/issues (project for opensuse infra issues)?

Reply:
I commented in https://github.com/openSUSE/zypper/issues/399#issuecomment-1386668065. As https://progress.opensuse.org/issues/123172 was resolved (on 2023-01-26) , I will monitor a few more builds before I open a new ticket.

I plan to raise up this kind of infra issue after we fixed some unstable issues from our end firstly. The tests will show more realiable results at that time.

Reply:
Sounds good.

Reply:
And I found things are not just on the opensuse side, in some cases, TW kernel downloads failed only in SUTs in O3 VLAN. I need more cases to confirm. I'll open a ticket to the right component when I get time to have it done.

Reply:
Posted a comment on https://progress.opensuse.org/issues/123328. Will keep an eye on it and determine what to do next.